lib/sanbase/external_services/etherscan/requests/internal_tx.ex

```
@@ -25,6 +25,8 @@ defmodule Sanbase.ExternalServices.Etherscan.Requests.InternalTx do
         error = "Error fetching transactions for #{address}. Status code: #{status}: #{body}"
         Logger.warn(error)
         {:error, error}
+      error ->
```

To match exactly the tesla error you can pattern match on this `%Tesla.Error{} = error ->` and stringify the error, as otherwise the recipient will not be able to print it. Other types of results are not expected (maybe not possible, too) and can be left to crash the process and generate a sentry log.

---

👍 The other solution may be to just return the `message` from `%Tesla.Error{}` here: https://github.com/santiment/sanbase2/pull/337/files#diff-34347235b130678fdce1211703e8ead9R8 to become: `e in Tesla.Error -> e.message`
@IvanIvanoff what do you think ?

---

Just `error` will match everything that didn't match the first two results and we won't be sure if that is some correct case that we missed or an error. Thus, it's not clear how it should be treated. My approach would be to treat the errors explicitly and either have a last case that catches everything and logs it as unrecognized message or just let it crash.

---

Yes, sound good. Implemented it that way